src/nvim/ex_getln.c

```diff
@@ -4308,11 +4306,12 @@ in_history (
       last_i = i;
     }
     history[type][i].hisnum = ++hisnum[type];
-    history[type][i].viminfo = FALSE;
     history[type][i].hisstr = str;
-    return TRUE;
+    history[type][i].timestamp = os_time();
+    history[type][i].additional_elements = NULL;
```

Comment:
@oni-link This is intentional. “Moved” entry is assumed to be semantically a new entry, so additional_elements list is not kept. I guess this code did not produce a memory leak only due to `garbagecollect()` calls and list becoming unreferenced due to history item being overwritten.